### PR TITLE
Export Fastq analyses to Trailblazer

### PR DESCRIPTION
**Pull request by islean**
## Description

Addresses #2832. We add a "dummy" analysis for Fastq cases with status "Completed" to trailblazer. Part of the user story https://github.com/Clinical-Genomics/cigrid-ui/issues/478.

### Changed

- Fastq cases now get an analysis set in Trailblazer with status Completed.

### CODE REVIEW

File: cg/cli/workflow/fastq/base.py [hunk 1]
```diff
@@ -39,6 +42,17 @@ def store_fastq_analysis(context: click.Context, case_id: str, dry_run: bool = F
...
     status_db.session.add(new_analysis)
     status_db.session.commit()
+    trailblazer_api: TrailblazerAPI = context.obj.trailblazer_api
+    trailblazer_api.add_pending_analysis(
+        case_id=case_id,
+        analysis_type=AnalysisType.OTHER,
+        data_analysis=Pipeline.FASTQ,
+        config_path="",
+        out_dir="",
+        slurm_quality_of_service=Priority.priority_to_slurm_qos().get(case_obj.priority),
```
Comment (islean) [inline cg/cli/workflow/fastq/base.py:52]: If these should be changed, please let me know. As far as we could tell, there was no need to set them, given that no actual analysis is run for a Fastq case.

File: cg/cli/workflow/fastq/base.py [hunk 2]
```diff
@@ -27,18 +30,29 @@ def store_fastq_analysis(context: click.Context, case_id: str, dry_run: bool = F
     """Creates an analysis object in status-db for the given fast case"""
     LOG.info(f"Creating an analysis for case {case_id}")
     status_db: Store = context.obj.status_db
-    case_obj: Case = status_db.get_case_by_internal_id(internal_id=case_id)
+    case: Case = status_db.get_case_by_internal_id(case_id)
```
Comment (ChrOertlin) [inline cg/cli/workflow/fastq/base.py]: ❤️

File: cg/cli/workflow/fastq/base.py [hunk 3]
```diff
@@ -27,18 +30,29 @@ def store_fastq_analysis(context: click.Context, case_id: str, dry_run: bool = F
...
         return
     status_db.session.add(new_analysis)
     status_db.session.commit()
+    trailblazer_api: TrailblazerAPI = context.obj.trailblazer_api
+    trailblazer_api.add_pending_analysis(
+        case_id=case_id,
+        analysis_type=AnalysisType.OTHER,
+        data_analysis=Pipeline.FASTQ,
+        config_path="",
+        out_dir="",
```
Comment (islean) [inline cg/cli/workflow/fastq/base.py:51]: These should be set to fetch the jobs

File: cg/cli/workflow/fastq/base.py [hunk 4]
```diff
@@ -27,18 +30,29 @@ def store_fastq_analysis(context: click.Context, case_id: str, dry_run: bool = F
     """Creates an analysis object in status-db for the given fast case"""
```
Comment (diitaz93) [inline cg/cli/workflow/fastq/base.py:30]:
```suggestion
    """Creates an analysis object in status-db for the given fast case."""
```

File: cg/cli/workflow/fastq/fastq_service.py
Comment (seallard) [inline cg/cli/workflow/fastq/fastq_service.py:1]: Maybe this file belongs in a different module and should be named differently

**Review by ChrOertlin — APPROVED** [review]
💯

**Review by diitaz93 — APPROVED** [review]
Great 💯

File: cg/cli/workflow/fastq/base.py [hunk 5]
```diff
@@ -23,22 +22,17 @@ def fastq(context: click.Context):
...
-        started_at=dt.datetime.now(),
-        case_id=case_obj.id,
-    )
+
     if dry_run:
         return
-    status_db.session.add(new_analysis)
-    status_db.session.commit()
+
+    fastq_service = FastqService(
```
Comment (seallard) [inline cg/cli/workflow/fastq/base.py:31]: This should not be instantiated here but rather in the app config

File: cg/cli/workflow/fastq/utils.py [hunk 6]
```diff
@@ -0,0 +1,7 @@
+from pathlib import Path
+
+import click
+
+
+def get_config_path(context: click.Context) -> str:
+    return Path(context.obj.data_delivery.base_path, "slurm_job_ids.yaml").as_posix()
```
Comment (Vince-janv) [inline cg/cli/workflow/fastq/utils.py:7]: What do we need this path for? Nothing will be created there right?

Comment (islean) [reply cg/cli/workflow/fastq/utils.py:7]: That path is wrong, but the config path is what we use to find the submitted jobs, right? Or is that via some other parameter? @Vince-janv

Comment (seallard) [reply cg/cli/workflow/fastq/utils.py:7]: Yes, we need a path to the the `slurm_job_ids.yaml` to track the rsync job. @beatrizsavinhas wants to be able to track the upload job for the fastq case in the new tab.

Comment (seallard) [reply cg/cli/workflow/fastq/utils.py:7]: The rsync dirs are named inconsistently, some use the database id and others the internal id. For example `271228_240108_10_53_05_784914` vs `betterweevil_240108_23_18_09_159755` which seems super annoying

Comment (seallard) [reply cg/cli/workflow/fastq/utils.py]: We will skip the upload job and fix separately for all pipelines and remove the rsync tab in trailblazer.

File: cg/cli/workflow/fastq/base.py (hunk 3 above)
Comment (seallard) [reply cg/cli/workflow/fastq/base.py]: Skipping for now, see below.

File: cg/cli/workflow/fastq/base.py (hunk 1 above)
Comment (islean) [reply cg/cli/workflow/fastq/base.py]: Out of scope.

**Pull request merged**